test/test_unary_ufuncs.py

```diff
@@ -137,6 +132,37 @@ def generate_numeric_tensors(device, dtype, *,
 
     return chain(empty_tensors, scalar_tensors, small_tensors, (medium_tensor,), (large_tensor,))
 
+
+def generate_numeric_tensors_hard(device, dtype):
+    is_signed_integral = dtype in (torch.int8, torch.int16, torch.int32, torch.int64)
+    assert dtype.is_floating_point or dtype.is_complex or is_signed_integral
+
+    if dtype.is_floating_point:
+        vals = _large_float_vals
+    elif dtype.is_complex:
+        vals = tuple(complex(x, y) for x, y in chain(product(_large_float_vals, _large_float_vals),
+                                                     product(_float_vals, _large_float_vals),
+                                                     product(_large_float_vals, _float_vals)))
+    else:
+        vals = _large_int_vals
+
+    return [torch.tensor(vals, device=device, dtype=dtype)]
+
+
+def generate_numeric_tensors_extremal(device, dtype):
+    assert dtype.is_floating_point or dtype.is_complex
+
+    vals = []
+    if dtype.is_floating_point:
+        vals = _float_extremals
+    elif dtype.is_complex:
+        vals = tuple(complex(x, y) for x, y in chain(product(_float_extremals, _float_extremals),
+                                                     product(_float_vals, _float_extremals),
+                                                     product(_float_extremals, _float_vals)))
+
+    return [torch.tensor(vals, device=device, dtype=dtype)]
```

Comment:
Same comment here about many scalar tensor to test non-vectorized code paths + one larger vectorized tensor.

Reply:
fixed